lib/adagrams.rb

```diff
@@ -0,0 +1,139 @@
+require 'csv'
+
+def return_letter_pool_hash
+  
+  letter_pool_hash = {
+  "A" => {number_of_tiles: 9, number_of_points: 1},
+  "B" => {number_of_tiles: 2, number_of_points: 3},
+  "C" => {number_of_tiles: 2, number_of_points: 3},
+  "D" => {number_of_tiles: 4, number_of_points: 2},
+  "E" => {number_of_tiles: 12, number_of_points: 1},
+  "F" => {number_of_tiles: 2, number_of_points: 4},
+  "G" => {number_of_tiles: 3, number_of_points: 2},
+  "H" => {number_of_tiles: 2, number_of_points: 4},
+  "I" => {number_of_tiles: 9, number_of_points: 1},
+  "J" => {number_of_tiles: 1, number_of_points: 8},
+  "K" => {number_of_tiles: 1, number_of_points: 5},
+  "L" => {number_of_tiles: 4, number_of_points: 1},
+  "M" => {number_of_tiles: 2, number_of_points: 3},
+  "N" => {number_of_tiles: 6, number_of_points: 1},
+  "O" => {number_of_tiles: 8, number_of_points: 1},
+  "P" => {number_of_tiles: 2, number_of_points: 3},
+  "Q" => {number_of_tiles: 1, number_of_points: 10},
+  "R" => {number_of_tiles: 6, number_of_points: 1},
+  "S" => {number_of_tiles: 4, number_of_points: 1},
+  "T" => {number_of_tiles: 6, number_of_points: 1},
+  "U" => {number_of_tiles: 4, number_of_points: 1},
+  "V" => {number_of_tiles: 2, number_of_points: 4},
+  "W" => {number_of_tiles: 2, number_of_points: 4},
+  "X" => {number_of_tiles: 1, number_of_points: 8},
+  "Y" => {number_of_tiles: 2, number_of_points: 4},
+  "Z" => {number_of_tiles: 1, number_of_points: 10}
+}
+
+return letter_pool_hash
+
+end
+
+def draw_letters
+  
+  letter_pool_hash = return_letter_pool_hash
+  
+  letter_pool_array = letter_pool_hash.map do |letter, letter_info|
+    (letter * letter_info[:number_of_tiles]).split("")
+  end
+  
+  letter_pool_array = letter_pool_array.flatten
+  
+  drawn_letters = letter_pool_array.sample(10)
+  return drawn_letters
+  
+end
+
+def uses_available_letters? (input, letters_in_hand)
+  
+  copy_of_hand = letters_in_hand.clone
+  letters_of_input = input.upcase.split("")
+  
+  letters_of_input.each do |letter_from_input|
+    if copy_of_hand.include? letter_from_input
+      
+      index_of_input_letter = copy_of_hand.index(letter_from_input)
+      copy_of_hand.delete_at(index_of_input_letter)
+      
+    else
+      
+      copy_of_hand = letters_in_hand.clone
```

Comment:
This method looks great! I just wanted to name that this line isn't necessary as the `copy_of_hand` variable as soon as the method returns on the next line. If that short explanation isn't clear, feel free to ask a fellow student or one of us instructors to get clarity on that.